packages/vite-plugin/src/partials/ffmpegBridge.ts

```diff
@@ -85,6 +86,89 @@ export function ffmpegBridgePlugin({output}: ExporterPluginConfig): Plugin {
           }
         },
       );
+
+      server.middlewares.use(
```

Comment:
There is a lot of repeated code here 😅. Claude gave me this which looks like a good starting point:
```ts
import {
  FFmpegExporterServer,
  FFmpegExporterSettings,
  VideoFrameExtractor,
  generateAudio,
  mergeMedia,
} from '@revideo/ffmpeg';
import {existsSync, unlinkSync} from 'fs';
import type {Plugin, WebSocketServer} from 'vite';

// ... existing code ...

export function ffmpegBridgePlugin({output}: ExporterPluginConfig): Plugin {
  return {
    name: 'revideo/ffmpeg',

    configureServer(server) {
      const ffmpegBridge = new FFmpegBridge(server.ws, {output});

      const handlePostRequest = async (req, res, handler) => {
        if (req.method !== 'POST') {
          res.statusCode = 405;
          res.end('Method Not Allowed');
          return;
        }

        const body = await new Promise((resolve, reject) => {
          let data = '';
          req.on('data', chunk => (data += chunk));
          req.on('end', () => resolve(data));
          req.on('error', reject);
        });

        try {
          const result = await handler(JSON.parse(body));
          res.statusCode = 200;
          if (result) {
            res.setHeader('Content-Type', 'application/json');
            res.end(JSON.stringify(result));
          } else {
            res.end('OK');
          }
        } catch (e) {
          console.error(e);
          res.statusCode = 500;
          res.end('Internal Server Error');
        }
      };

      server.middlewares.use('/audio-processing/generate-audio', (req, res) =>
        handlePostRequest(req, res, async ({tempDir, assets, startFrame, endFrame, fps}) =>
          generateAudio({outputDir: output, tempDir, assets, startFrame, endFrame, fps})
        )
      );

      server.middlewares.use('/audio-processing/merge-media', (req, res) =>
        handlePostRequest(req, res, async ({outputFilename, tempDir}) =>
          mergeMedia(outputFilename, output, tempDir)
        )
      );

      server.middlewares.use('/revideo-ffmpeg-decoder/video-frame', (req, res) =>
        handlePostRequest(req, res, async (data) => {
          const {frame, width, height} = await ffmpegBridge.handleDecodeVideoFrame(data);
          res.setHeader('X-Frame-Width', width.toString());
          res.setHeader('X-Frame-Height', height.toString());
          res.setHeader('Content-Type', 'application/octet-stream');
          res.end(Buffer.from(frame as Buffer));
        })
      );

      server.middlewares.use('/revideo-ffmpeg-decoder/finished', (req, res) =>
        handlePostRequest(req, res, () => ffmpegBridge.handleRenderFinished())
      );

      server.middlewares.use('/revideo-ffmpeg-decoder/download-video-chunks', (req, res) =>
        handlePostRequest(req, res, async (videoDurations) => {
          const downloadedPaths = await ffmpegBridge.handleDownloadVideoChunks(videoDurations);
          return {success: true, paths: downloadedPaths};
        })
      );
    },
  };
}
```